docs/intentReaderHandlers/RegisterProviderHandler.md

```diff
@@ -0,0 +1,187 @@
+
+* Register for the Keyboard Test Provider by sending the below intent - src/providers/KeyboardUIProviderTest.js
+*  
+```json
+
+    {
+        "action": "search",
+        "data": {
+            "query": {
+                "task": "registerProviderHandler",
+                "params": {
+                    "provider": "keyboard",
+                    "params": []
+                },
+                "action": "NA",
+                "appType": "firebolt"
+            }
+        },
+        "context": {
+            "source": "device"
+        }
+    }
+
+```
+* Set the response for keyboard.email using the SetApiResponseHandler by sending the below intent
+*   
+```json
+
+    {
+        "action": "search",
+        "data": {
+            "query": {
+                "task": "setApiResponse",
+                "params": {
+                    "apiResponse": {
+                        "module": "keyboard",
+                        "methodName": "keyboard.email",
+                        "type": "method",
+                        "attributes": [
+                            {
+                                "ApiText": "abc@gmail.com",
+                                "isCancelled": false,
+                                "withUi": false,
+                                "result": "john@doe.com"
+                            }
+                        ]
+                    }
+                },
+                "action": "NA",
+                "appType": "firebolt"
+            }
+        },
+        "context": {
+            "source": "device"
+        }
+    }
+
+```
+* Call Keyboard.email api from UI .
+* With the intents sent to RegisterProviderHandler and SetApiResponseHandler , keyboard.email will return abc@gmail.com
+
+### Request Format
+
+```json
+    {
+        "action": "search",
+        "data": {
+            "query": {
+                "task": "registerProviderHandler",
+                "params": {
+                    "provider": "<providerName>",
+                    "params": []
+                },
+                "action": "NA",
+                "appType": "<appType>"
+            }
+        },
+        "context": {
+            "source": "device"
+        }
+    }
+```
+
+#### Parameters
+
+| Key               | Description                                                                                   | Required? |
+|-------------------|-----------------------------------------------------------------------------------------------|-----------|
+| task              | "registerProviderHandler"- It is a static value and should not be changed for this handler      | Y         |
+| params            | Required provider params for  the intent. Here, "provider" is a mandatory parameter           | Y         |
+| appType           | Corresponding intent is launching on which app                                                | Y         |
+
+### Response Format
+* Response can be either "true" or an error response
```

Comment:
So just want to make sure this is accurate because in the example valid response you have a "success message" instead of a "true". Please make sure the following things are all in sync with each other:
- `Response Format` description
- Response Parameters
- Example Responses

Please make sure these are aligned in all the handlers for both positive and negative scenarios, I think the majority are good already but have seen a few that looks out of place.

Reply:
updated the responses